partiql-lang/src/main/kotlin/org/partiql/lang/syntax/impl/PartiQLPigVisitor.kt

```diff
@@ -1507,6 +1510,12 @@ internal class PartiQLPigVisitor(
      */
 
     override fun visitBag(ctx: PartiQLParser.BagContext) = PartiqlAst.build {
+        // Prohibit hidden characters between angle brackets
+        val startTokenIndex = ctx.start.tokenIndex
+        val endTokenIndex = ctx.stop.tokenIndex
+        if (tokens.getHiddenTokensToRight(startTokenIndex, PartiQLTokens.HIDDEN) != null || tokens.getHiddenTokensToLeft(endTokenIndex, PartiQLTokens.HIDDEN) != null) {
+            throw ParserException("Invalid bag expression", ErrorCode.PARSE_INVALID_QUERY)
```

Comment:
https://github.com/partiql/partiql-lang-kotlin/blob/4dd09721d56cacf19f4b7e312ac927f247a20975/partiql-lang/src/test/kotlin/org/partiql/lang/syntax/PartiQLParserTestBase.kt#L164-L166

Here if I remembered correctly the error code has not been implemented for the experimental parser and hence it should be a not equal operation.

Reply:
I was a bit tripped up by this testing method's (`checkInputThrowingParserException`) logic. For clarity,
- ParserTarget.DEFAULT -> refers to parser for PIG AST
- ParserTarget.EXPERIMENTAL -> refers to parser for partiql-ast

The `targets` parameter of `checkInputThrowingParserException` is unused -- https://github.com/partiql/partiql-lang-kotlin/blob/4dd09721d56cacf19f4b7e312ac927f247a20975/partiql-lang/src/test/kotlin/org/partiql/lang/syntax/PartiQLParserTestBase.kt#L155. Instead, the `targets` property defined for the abstract parser test class is used https://github.com/partiql/partiql-lang-kotlin/blob/4dd09721d56cacf19f4b7e312ac927f247a20975/partiql-lang/src/test/kotlin/org/partiql/lang/syntax/PartiQLParserTestBase.kt#L38.

This function currently just does two things:
- for `ParserTarget.EXPERIMENTAL`, check for an error and validate the error code + error context
- for `ParserTarget.DEFAULT`, check for an error and do **NO** further validations.

That is why the error codes differ between the two error tests and why the error tests use the `ParserTarget.EXPERIMENTAL` error code, `PARSE_UNEXPECTED_TOKEN`.

Reply:
Since we're going to get rid of `ParserTarget.DEFAULT` for `v1`, I didn't want to alter the testing infrastructure further. I'll leave the tests as they are and add a couple comments for clarity.